### ISSUE TITLE
AR::Base#reload and ignored_columns

### ISSUE THREAD

**Issue by kirs** [rails/rails#27914]
The `ignored_columns` list is ignored after you do `reload` on a record:

On master:

```ruby
require 'bundler/setup'
require 'active_record'
require "minitest/autorun"
require "logger"

ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table "users" do |t|
    t.string "name"
    t.integer "age"
  end
end

class User < ActiveRecord::Base
  self.ignored_columns = ["age"]
end

class BugTest < Minitest::Test
  def test_association_stuff
    user = User.create!
    refute user.respond_to?(:age) # passes
    refute user.respond_to?(:age=) # passes
    user.reload
    refute user.respond_to?(:age) # test fails here because `age` is not available
  end
end
```

I guess this happens because on reload we do `SELECT *`.
Do we consider it as a bug?

@sgrif @matthewd

**Comment by steverob** [comment]
Inside `reload`, after `@attributes` are reset [here](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/persistence.rb#L457), the ignored columns stop working. 

Been trying to dig in and find out how this happens, no luck.

**Comment by guilleiguaran** [comment]
@kirs confirm, this is a bug

**Comment by maclover7** [comment]
Making good headway towards a solution.

r? @maclover7

**Comment by maclover7** [comment]
Opened #27918 to try and solve this.

**Comment by sobrinho** [comment]
@rafaelfranca can be closed in favor of #30980.
